weather.py

```diff
@@ -83,6 +134,21 @@ def find_max(weather_data):
 
 
 def generate_summary(weather_data):
+    summary= ""
+    low = []
+    high = []
+    for row in weather_data:
+        low.append(row[1])
+        high.append(row[2])
+
+    summary = f"{len(weather_data)} Day Overview\n\
+  The lowest temperature will be {format_temperature(convert_f_to_c(find_min(low)[0]))}, and will occur on {convert_date(weather_data[find_min(low)[1]][0])}.\n\
```

Comment:
Nice use of f-strings.

You could also use tuple unpacking as you did earlier, so that you can avoid referencing the list index [0] and [1]. E.g.
```suggestion
min_temp, min_temp_idx = find_min(low)
    summary = f"{len(weather_data)} Day Overview\n\
  The lowest temperature will be {format_temperature(convert_f_to_c(min_temp))}, and will occur on {convert_date(weather_data[find_min(low)[1]][0])}.\n\
  The highest temperature will be {format_temperature(convert_f_to_c(min_temp_idx))}, and will occur on {convert_date(weather_data[find_max(high)[1]][0])}.\n\
```